wgengine/router/router_func.go

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) 2020 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package router
+
+// FuncRouter is a Router to the functions it embeds.
+// It is useful, for example, to pass configs into ipn-go-bridge on macOS/iOS.
```

Comment:
// It's used by the macOS/iOS apps.

Actually, maybe this code just belongs in that repo.

No reason for it to be here now, is there?

Reply:
The only reason is really `shutdownConfig`, which conceptually is an implementation detail of the `router` package, I guess. The right thing to do would probably be moving it up a level.

Reply:
Well, shutdownConfig is a little weird anyway. It's just the zero value. Will it ever not be the zero value?

We could make SetConfig take a value and not a pointer and then shutdownConfig goes away if we document that the Config zero value means "remove everything"?

Reply:
I _think_ it was only made a pointer because it was reasonably large, so `shutdownConfig` should always be the zero value. I guess saying "nil argument means everything is removed" is also reasonable, and this is in effect the current contract, with `shutdownConfig` being an implementation detail. I've kept the interface as-is for now; it's not too bad and mucking around with this has stretched this out more than it probably should have.